server/neptune/gateway/event/push_event_handler.go

```diff
@@ -149,18 +160,70 @@ func (p *PushHandler) generatePlanSteps(repoID string) []workflows.Step {
 	return workflowSteps
 }
 
-func (p *PushHandler) generateApplySteps(repoID string) []workflows.Step {
-	var workflowSteps []workflows.Step
-	projectConfig := p.GlobalCfg.DefaultProjCfg(p.Logger, repoID, "path", defaultWorkspace)
-	steps := projectConfig.Workflow.Apply.Steps
-	for _, step := range steps {
-		workflowSteps = append(workflowSteps, workflows.Step{
-			StepName:    step.StepName,
-			ExtraArgs:   step.ExtraArgs,
-			RunCommand:  step.RunCommand,
-			EnvVarName:  step.EnvVarName,
-			EnvVarValue: step.EnvVarValue,
-		})
+func (p *PushHandler) buildRoots(event Push, ctx context.Context) ([]*valid.MergedProjectCfg, error) {
+	err := p.PreWorkflowHooksCommandRunner.RunPreHooksWithSha(ctx, p.Logger, event.Repo, event.Sha)
+	if err != nil {
+		p.Logger.Error(fmt.Sprintf("Error running pre-workflow hooks %s. Proceeding with root building.", err))
 	}
-	return workflowSteps
+
+	modifiedFiles, err := p.VCSClient.GetModifiedFilesFromCommit(event.Repo, event.Sha)
+	if err != nil {
+		return nil, err
+	}
+
+	unlockFn, err := p.WorkingDirLocker.TryLockOnSha(event.Repo.FullName, event.Sha, DefaultWorkspace)
```

Comment:
I'm of the opinion we shouldn't support workspace if we can either. it's a concept that hashicorp says shouldn't even be used for production systems. (Terraform cloud doesn't even use this concept).